app/src/ui/rebase/choose-branch.tsx

```diff
@@ -154,15 +181,115 @@ export class ChooseBranchDialog extends React.Component<
     )
   }
 
+  private renderRebaseStatus = () => {
+    const { currentBranch, rebasePreviewStatus } = this.props
+    const { selectedBranch } = this.state
+
+    if (rebasePreviewStatus === null) {
+      return null
+    }
+
+    if (selectedBranch === null) {
+      return null
+    }
+
+    if (currentBranch.name === selectedBranch.name) {
+      return null
+    }
+
+    return (
+      <div className="rebase-status-component">
+        <ActionStatusIcon
+          status={this.props.rebasePreviewStatus}
+          classNamePrefix="rebase-status"
+        />
+        <p className="rebase-message">
+          {this.renderRebaseDetails(
+            currentBranch,
+            selectedBranch,
+            rebasePreviewStatus
+          )}
+        </p>
+      </div>
+    )
+  }
+
+  private renderRebaseDetails(
+    currentBranch: Branch,
+    baseBranch: Branch,
+    rebaseStatus: RebasePreview
+  ): JSX.Element | null {
+    if (rebaseStatus.kind === ComputedAction.Loading) {
+      return this.renderLoadingRebaseMessage()
+    }
+    if (rebaseStatus.kind === ComputedAction.Clean) {
+      return this.renderCleanRebaseMessage(
+        currentBranch,
+        baseBranch,
+        rebaseStatus.commits.length
+      )
+    }
+
+    // TODO: other scenarios to display some context about
+
+    return null
+  }
+
+  private renderLoadingRebaseMessage() {
+    return (
+      <React.Fragment>
```

Comment:
Seems like we've used `<> ... </>` in most places in the app, could we change that here for consistency or even better just return a string here?